### PR TITLE
Backport/move compatibility

### PR DESCRIPTION
**Pull request by xavierh**
Depends of #1152 
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/util/Utils.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.minecolonies.coremod.util;
 
+import com.minecolonies.api.compatibility.Compatibility;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/Utils.java]: MINOR Remove this unused import 'com.minecolonies.api.compatibility.Compatibility'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 75 issues
* ![MAJOR][MAJOR] 24 major
* ![MINOR][MINOR] 38 minor
* ![INFO][INFO] 13 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L60](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L60): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L780](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L780): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L51](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L176](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L176): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkFarmer.java#L124](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java#L124): The Cyclomatic Complexity of this method "prepareForFarming" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkFarmer.java#L474](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java#L474): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIMeleeGuard.java#L116](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java#L116): Split this 181 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [Tree.java#L317](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java#L317): Remove this unused private "checkTree" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [Structure.java#L65](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L65): Reduce this class from 47 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [AbstractPathJob.java#L31](https://github.com/Minecolonies/minecolonies/blob/a525d4a9b6624e9c0f33d75e47f85b5ead418168/src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractPathJob.java#L31): class "AbstractPathJob" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 73 issues
* ![MAJOR][MAJOR] 24 major
* ![MINOR][MINOR] 37 minor
* ![INFO][INFO] 12 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L60](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L60): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L780](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L780): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L51](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L176](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L176): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkFarmer.java#L124](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java#L124): The Cyclomatic Complexity of this method "prepareForFarming" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkFarmer.java#L474](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java#L474): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIMeleeGuard.java#L116](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java#L116): Split this 181 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [Tree.java#L317](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java#L317): Remove this unused private "checkTree" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [Structure.java#L65](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L65): Reduce this class from 47 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [AbstractPathJob.java#L31](https://github.com/Minecolonies/minecolonies/blob/1a5b4e47458b32ab8f01fed6b64d835aa31832ce/src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractPathJob.java#L31): class "AbstractPathJob" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: as well

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1153/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
@xavierh this here has some conflicts

**Review by marchermans — APPROVED** [review]

**Comment by Raycoms** [comment]
:shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 34 minor
* ![INFO][INFO] 4 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIMeleeGuard.java#L116](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java#L116): Split this 181 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [Tree.java#L317](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java#L317): Remove this unused private "checkTree" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/6fabd7574bc443389efa8744ebaff514f5c940d0/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**